src/css/cssCastUnit.js

```diff
@@ -0,0 +1,60 @@
+define([
+	"../core",
+	"../var/rfxnum"
+], function( jQuery, rfxnum ) {
+
+function cssCastUnit( elem, prop, parts, tween ) {
+	var getCSS = tween ?
+			function() { return tween.cur(); } :
+			function() { return jQuery.css( elem, prop, "" ); },
+		target = getCSS(),
+		unit = parts && parts[ 3 ] || ( jQuery.cssNumber[ prop ] ? "" : "px" ),
+		// Starting value computation is required for potential unit mismatches
+		start = ( jQuery.cssNumber[ prop ] || unit !== "px" && +target ) &&
+			rfxnum.exec( jQuery.css( elem, prop ) ),
+		end,
+		scale = 1,
+		maxIterations = 20;
+
+	if ( start && start[ 3 ] !== unit ) {
+		// Trust units reported by jQuery.css
+		unit = unit || start[ 3 ];
+
+		// Make sure we update the tween properties later on
+		parts = parts || [];
+
+		// Iteratively approximate from a nonzero starting point
+		start = +target || 1;
+
+		do {
+			// If previous iteration zeroed out, double until we get *something*.
+			// Use string for doubling so we don't accidentally see scale as unchanged below
+			scale = scale || ".5";
+
+			// Adjust and apply
+			start = start / scale;
+			jQuery.style( elem, prop, start + unit );
+
+			// Update scale, tolerating zero or NaN from tween.cur(),
+			// break the loop if scale is unchanged or perfect, or if we've just had enough
```

Comment:
These comments are indented one tab too far. Also, this is the perfect opportunity to fix my bad grammar by removing the trailing comma and capitalizing "break".